apisix/plugins/openid-connect.lua

```diff
@@ -374,18 +383,34 @@ local function introspect(ctx, conf)
             return nil, nil, nil, nil
         end
     end
-
+    local opts = {}
     -- If we get here, token was found in request.
-
+    if conf.use_jwks then
+        local valid_issuers
+        if conf.valid_issuers then
+            valid_issuers = conf.valid_issuers
+        else
+            local discovery, discovery_err = openidc.get_discovery_doc(conf)
+            if discovery_err then
+                core.log.warn("OIDC access discovery url failed : ", discovery_err)
+            else
+                core.log.info("valid_issuers not provided, using issuer from discovery doc: ",
+                              discovery.issuer)
+                valid_issuers = {discovery.issuer}
+            end
+        end
+        if valid_issuers then
+            opts.valid_issuers = valid_issuers
+        end
+    end
     if conf.public_key or conf.use_jwks then
```

Comment:
If public_key is specified, it will not verify issuers. Are you sure this is expected?

Reply:
you are right, we should verify issuer when use `public_key` too.